master/docs/manual/cfg-global.rst

```diff
@@ -276,27 +276,39 @@ Data Lifetime
 .. bb:cfg:: changeHorizon
 .. bb:cfg:: buildHorizon
 .. bb:cfg:: logHorizon
+.. bb:cfg:: JanitorConfigurator
 
 Horizons
 ++++++++
 
 ::
 
-    c['changeHorizon'] = 200
-    c['buildHorizon'] = 100
-    c['logHorizon'] = 40
-    c['buildCacheSize'] = 15
+    from buildbot.plugins import util
+    from datetime import timedelta
+
+    # at the end of your master.cfg
+    # configure a janitor which will delete all logs older than one month, and will run on sundays at noon
+    util.JanitorConfigurator(c,
+        logHorizon=timedelta(weeks=4),
+        hour=12,
+        dayOfWeek=6)
 
 Buildbot stores historical information in its database.
 In a large installation, these can quickly consume disk space, yet in many cases developers never consult this historical information.
 
-The :bb:cfg:`changeHorizon` key determines how many changes the master will keep a record of.
-One place these changes are displayed is on the waterfall page.
-This parameter defaults to 0, which means keep all changes indefinitely.
+Parameters for :bb:cfg:`JanitorConfigurator` are:
+
+``c``
```

Comment:
Adding another global params key seems like it would fit in well with the rest of the Buildbot configuration. Everything else uses that except for configurators.

I can't comment on ramifications of running on the main reactor thread.